### PR TITLE
Use per-team LLM Provider configuration for LLM interactions

### PR DESCRIPTION
**Pull request by snopoke**
Review by commit.

With this change the global OpenAI API key will be unused and all interactions will use the API key configured for the experiment.

### CODE REVIEW

File: apps/teams/context_processors.py [hunk 1]
```diff
@@ -1,4 +1,33 @@
...
+        "notices": get_team_notices(request),
     }
+
+
+def get_team_notices(request):
+    team = getattr(request, "team", None)
+    if not team:
+        return []
+
+    return filter(None, [notice(request, team) for notice in NOTICES])
```
Comment (SmittieC) [inline apps/teams/context_processors.py:16]: Why the `filter(None, ...)`? Wouldn't this just return the iterable?

File: apps/experiments/migrations/0047_bootstrap_llm_provider.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    Team = apps.get_model("teams", "Team")
+    LlmProvider = apps.get_model("llm_providers", "LlmProvider")
+    Experiment = apps.get_model("experiments", "Experiment")
+
+    for team in Team.objects.all():
+        provider = LlmProvider.objects.filter(team=team).first()
+        if not provider:
+            continue
+
+        Experiment.objects.filter(team=team).update(llm_provider=provider)
```
Comment (SmittieC) [inline apps/experiments/migrations/0047_bootstrap_llm_provider.py:16]: Is my understanding correct in that there will exist no `LlmProvider` records on prod yet, and that when this code is deployed, we will need to create one and link them to the experiments after these changes?

Comment (SmittieC) [reply apps/experiments/migrations/0047_bootstrap_llm_provider.py:16]: If so, that would mean that this migration should be run _after_ that step. Maybe I'm missing something

File: apps/teams/context_processors.py (hunk 1 above)
Comment (snopoke) [reply apps/teams/context_processors.py:16]: It removes any that are 'falsy'

File: apps/experiments/migrations/0047_bootstrap_llm_provider.py (hunk 2 above)
Comment (snopoke) [reply apps/experiments/migrations/0047_bootstrap_llm_provider.py:16]: Correct. I've already created the LlmProvider on prod

File: apps/teams/context_processors.py (hunk 1 above)
Comment (SmittieC) [reply apps/teams/context_processors.py:16]: Cool, did not know that!

File: apps/experiments/migrations/0047_bootstrap_llm_provider.py (hunk 2 above)
Comment (SmittieC) [reply apps/experiments/migrations/0047_bootstrap_llm_provider.py:16]: Coolios

**Review by SmittieC — APPROVED** [review]
Looks good to me

**Pull request merged**